tests/unit/algos/list/delimiterMaster.spec.js

```diff
@@ -488,6 +600,92 @@ function getListingKey(key, vFormat) {
+                            // const prefix = baseKey.slice(0, lastIdx); 
+                            const prefix = getCommonPrefix(baseKey, '/', lastIdx);
+                            if (state.prefixes.has(prefix)) {
+                                return true;
+                            } else {
+                                state.prefixes.add(prefix)
+                                return canSkip;
+                            }
+                        }
+                    } else {
+                        state.handleSubprefixKey = (baseKey, canSkip) => {
+                            let lastIdx = state.prefix.length;
+                            const prefix = baseKey.slice(0, lastIdx);
+                            if (state.prefixes.has(prefix)) {
+                                return true;
+                            } else if (baseKey.length > lastIdx){
+                                return true;
+                            } else {
+                                state.prefixes.add(prefix)
+                                return canSkip;
+                            }
+                        }
+                    }
+                    
+                    const maxKeys = 1000000;
+                    const delimiter = new DelimiterMaster({
+                        maxKeys,
+                        prefix,
+                        delimiter: '/',
+                        startAfter: '',
+                        continuationToken: '',
+                        v2: true,
+                        fetchOwner: false }, fakeLogger, vFormat);
+                    
+                    [...generateKeys(maxKeys, state)].forEach(ob => {
+                        const {key, isDeleteMarker, canSkip} = ob;
+                        // simulate not doing a raw listing when key < params.gte
+                        // this represents a key not reaching the read stream from leveldb
+                        if (state.params.gte && key < state.params.gte) {
+                            return;
+                        }
+        
+                        if (!key.includes(VID_SEP)) { // master key
+                            const version = new Version({ isDeleteMarker });
+                            const obj = {
+                                key: getListingKey(key, vFormat),
+                                value: version.toString(),
+                            };
+        
+                            const res = delimiter.filter(obj);
+                            const skippingRange = delimiter.skipping();
+                            handleStreak(res, skippingRange, state);
+        
+                            let expected = null;
+                            if (vFormat === 'v0') {
+                                expected = canSkip ? FILTER_SKIP : FILTER_ACCEPT;
+                            } else { // v1
+                                expected = FILTER_ACCEPT
```

Comment:
I don't understand why in v1 all master keys are accepted. I would think that the behavior should be similar for v0 and v1 when it comes to master keys handling.